client/signup/steps/user/index.jsx

```diff
@@ -333,6 +333,16 @@ export class UserStep extends Component {
 			);
 		}
 
+		const redirectToAfterLoginUrl = getRedirectToAfterLoginUrl( this.props );
+
+		if ( redirectToAfterLoginUrl?.startsWith( '/setup/hosted-site-migration' ) ) {
+			subHeaderText = translate(
+				'Pick an option to start moving your site to the world’s best WordPress host.'
+			);
+		} else if ( redirectToAfterLoginUrl?.startsWith( '/start/do-it-for-me' ) ) {
```

Comment:
Do we already have an intent at this moment? Otherwise, this seems a good path to me.

Reply:
I think it depends on how you get it. This part is not too friendly to WP data stores.

Reply:
Yeah it's sorta dicey 😅 Really we want to know where they came _from_, not where they're _going_. But I think this is good enough if it's able to distinguish between the cases we need it to.